src/System.Memory/src/System/SpanHelpers.byte.cs

```diff
@@ -456,30 +428,22 @@ public static unsafe int IndexOfAny(ref byte searchSpace, byte value0, byte valu
                     }
                     // Found match, reuse Vector vComparison to keep register pressure low
```

Comment:
You should be able to remove this bit too and just use vMatches to compute the result.